platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/maplayout/RecyclerViewActivity.kt

```diff
@@ -0,0 +1,148 @@
+package com.mapbox.mapboxsdk.testapp.activity.maplayout
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.TextView
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.testapp.R
+import kotlinx.android.synthetic.main.activity_recyclerview.*
+
+/**
+ * TestActivity showcasing how to integrate a MapView in a RecyclerView.
+ * <p>
+ * It requires calling the correct lifecycle methods when detaching and attaching the View to
+ * the RecyclerView with onViewAttachedToWindow and onViewDetachedFromWindow.
+ * </p>
+ */
+@SuppressLint("ClickableViewAccessibility")
+class RecyclerViewActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_recyclerview)
+        recyclerView.layoutManager = LinearLayoutManager(this)
+        recyclerView.adapter = ItemAdapter(LayoutInflater.from(this), savedInstanceState)
+    }
+
+    override fun onSaveInstanceState(outState: Bundle?) {
+        super.onSaveInstanceState(outState)
+        // to save state, we need to call MapView#onSaveInstanceState
+        (recyclerView.adapter as ItemAdapter).onSaveInstanceState(outState)
+    }
+
+    override fun onLowMemory() {
+        super.onLowMemory()
+        // to release memory, we need to call MapView#onLowMemory
+        (recyclerView.adapter as ItemAdapter).onLowMemory()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        // to perform cleanup, we need to call MapView#onDestroy
+        (recyclerView.adapter as ItemAdapter).onDestroy()
```

Comment:
I feel like some of the problems mentioned above might stem from calling `MapView#onDestroy` only when activity is destroyed. We should call that whenever the `MapView` holder is destroyed? Maybe `#onViewRecycled`? Not sure here.

Reply:
The issue above was the view measurement from https://github.com/mapbox/mapbox-gl-native/pull/13133 . View recycling still keeps the View in memory, it's detached but keeps it alive. OnDestroy doesn't match that step at that time completely. Destroying at view recycle time would also result in recreating it completely while scrolling up and down the list. This results in hick-up during scrolling and seeing the black surface for a second. I initially went with that approach but was not satisfied with the experience.